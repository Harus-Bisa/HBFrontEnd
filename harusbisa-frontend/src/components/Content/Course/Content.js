import React from "react";
import StudentCourseCard from "../../Card/StudentCourseCard";
import {connect} from "react-redux";

function mapStateToProps(state){
    return{
        courses: state.courses
    }
}

function Content(props){
    const makeCourses = () =>{
        var courses = props.courses
        var components = []
        for (let i=0; i<courses.length; i++){
            var id = courses[i]._id
            components.push(<StudentCourseCard key={i} id={id}/>)
        }
        if (courses.length === 0){
            return(<p>Make your first course!</p>)
        }
        return components;
    }
    return(
        <div>
            <header>
                <h1>Kelas Anda</h1>
            </header>
            <div className="row">
                {makeCourses()}
            </div>
        </div>
    )
}

export default connect(mapStateToProps)(Content);