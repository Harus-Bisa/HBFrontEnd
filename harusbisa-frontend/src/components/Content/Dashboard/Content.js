import React from "react";
import {connect} from "react-redux";
import QuizCard from "../../Card/QuizCard";

function mapStateToProps(state){
    return {
        lecture: state.selectedLecture
    }
}
function Content(props){
    const makeQuizzes = () =>{
        var quizzes = []
        for (let i=0; i<props.lecture.quizzes.length; i++){
            quizzes.push(<QuizCard key={i} index={i}/>)
        }
        return quizzes;
    }
    if (!props.lecture){
        return(
            <div>Make your first lecture</div>
        )
    }
    return(
        <div>
            <header>
                <h1>Sesi {props.lecture.date}</h1>
            </header>
            <div>
                {makeQuizzes()}
            </div>
        </div>
    )
}

export default connect(mapStateToProps)(Content);
