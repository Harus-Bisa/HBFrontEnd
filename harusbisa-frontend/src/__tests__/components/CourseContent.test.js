import React from "react";
import TestRenderer  from "react-test-renderer";
import {shallow} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import Content from "../../components/Content/Course/Content";
import {Provider} from 'react-redux';

const mockStore = configureMockStore();
describe("CourseContent component", () =>{
    let initialState;
    let page;
    let store;
    beforeEach(() =>{
        initialState = {
            courses:[{
                "_id": "5db492c349a67b33b8d0a2a2",
                "course_name": "Calculus I",
                "course_description": "The basic of calculus (derivatives).",
                "start_term": "August 2019",
                "end_term": "December 2019",
                "number_of_lectures": 5,
                "instructor": "John Doe"
              }]
        }
        store = mockStore(initialState);
        page = shallow(<Provider store={store}><Content/></Provider>)
    })

    it("Matches snapshot", () =>{
        expect(page.html()).toMatchSnapshot();
    })
})