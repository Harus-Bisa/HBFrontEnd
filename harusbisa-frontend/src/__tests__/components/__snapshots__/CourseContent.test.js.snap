// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseContent component Matches snapshot 1`] = `"<div class=\\"content\\"><header><div class=\\"row\\"><div class=\\"col-6\\"><h1>Kelas Anda</h1></div><div class=\\"col-6 d-none d-md-block\\" style=\\"align-self:center\\"><div class=\\"row\\"><div class=\\"col-6\\" style=\\"display:flex;justify-content:flex-end\\"><div style=\\"margin:auto 0\\"><div class=\\"search-bar\\"><div class=\\"col-3\\" style=\\"display:flex\\"><svg class=\\"MuiSvgIcon-root icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></div><div class=\\"col-9\\"><input type=\\"text\\" placeholder=\\"Cari kelas\\"/></div></div></div></div><div class=\\"col-6\\" style=\\"display:flex;justify-content:flex-end\\"><div style=\\"margin:auto 0\\"><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text student-button\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">+ Tambah Kelas</span></button></div></div></div></div></div></div></header><div class=\\"row\\"><div class=\\"col-md-6\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root card student-card MuiPaper-rounded\\"><div class=\\"MuiCardHeader-root\\"><div class=\\"MuiCardHeader-content\\"><span class=\\"MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock\\"><h5><a href=\\"/student/dashboard/5db492c349a67b33b8d0a2a2\\">Calculus I</a></h5></span></div><div class=\\"MuiCardHeader-action\\"><div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span></button></div></div></div><div class=\\"MuiCardContent-root d-none d-md-block\\"><p>August 2019 - December 2019</p><p>John Doe</p></div><div class=\\"MuiCardActions-root footer MuiCardActions-spacing\\"><p>5 Sesi</p></div></div></div></div></div>"`;
